ui/packages/shared/functions/src/useUIFeatureFlag/index.ts

```diff
@@ -15,36 +15,39 @@ import useLocalStorageState from 'use-local-storage-state';
 
 const UI_FLAGS = 'ui-flags';
 
-const initializeFlagsFromURL = () => {
+const initializeFlagsFromURL = (): void => {
   if (typeof window === 'undefined') {
     return;
   }
 
   const url = new URL(window.location.href);
   const enableFlag = url.searchParams.get('enable-ui-flag');
   const disableFlag = url.searchParams.get('disable-ui-flag');
-  if (!enableFlag && !disableFlag) {
+  if (enableFlag !== null && disableFlag !== null) {
     return;
   }
   const flags = JSON.parse(window.localStorage.getItem(UI_FLAGS) ?? '{}');
-  if (enableFlag) {
+  if (enableFlag !== null) {
     flags[enableFlag] = true;
   }
-  if (disableFlag) {
+  if (disableFlag !== null) {
     delete flags[disableFlag];
   }
   window.localStorage.setItem(UI_FLAGS, JSON.stringify(flags));
 };
 
 initializeFlagsFromURL();
 
-const useUIFeatureFlag = (featureFlag: string, defaultValue: boolean = false) => {
+const useUIFeatureFlag = (
+  featureFlag: string,
+  defaultValue: boolean = false
+): Array<boolean | ((key: string) => boolean)> => {
```

Comment:
This return type does not look right to me, maybe:

```suggestion
): [boolean, ((flag: boolean) => void)] => {
```

Reply:
oh, good catch!

Reply:
oh, good catch!